.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,102 @@
+name: build_and_test
+
+# Only run tests for main branch or if the PR has relevant changes
+on: 
+  push:
+    branches:
+      - main
+  pull_request:
+
+concurrency:
+  group: build_and_test-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+env:
+  RESULT_PATH: testresults
+  PYTHON_VERSION: 3.8.10
+  REQUIREMENTS_PATH: "tests/requirements.txt"
+
+jobs:
+  build:
+    name: build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the codebase.
+        uses: actions/checkout@v2
+
+      - name: Setup bundle
+        env:
+          PULL_CACHE: yes
+          AGENT_VERSION: latest
+        run: |
+          make bundle
+          mkdir -p /tmp/workspace
+          docker save -o /tmp/workspace/image.tar quay.io/signalfx/signalfx-agent-dev:latest
+          mv signalfx-agent-latest.tar.gz /tmp/workspace/
+
+      - name: Uploading artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: bundle
+          path: /tmp/workspace/*
```

Comment:
Let's split the artifacts into separate steps, one named `image` for `image.tar` and the other named `bundle` for `signalfx-agent-latest.tar.gz` so that dependent jobs won't need to unnecessarily download both.

Reply:
As per your suggestion, I have changed the workflow to split the artifacts into separate steps.